### PR TITLE
upgrading sample-apiserver to 1.10

### LINKED ISSUE #63622: Flaky Test : [sig-api-machinery] Aggregator Should be able to support the 1.7 Sample API Server using the current Aggregator fails on "pull-kubernetes-e2e-gce" and "ci-kubernetes-e2e-gci-gce"
## Flaky Tests
* [pull-kubernetes-e2e-gce](https://k8s-gubernator.appspot.com/builds/kubernetes-jenkins/pr-logs/pull/59284/pull-kubernetes-e2e-gce/)
* [ci-kubernetes-e2e-gci-gce](https://k8s-gubernator.appspot.com/builds/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce/)

## Flakiest test 
[sig-api-machinery] Aggregator Should be able to support the 1.7 Sample API Server using the current Aggregator

## Failures in TestGrid
* [pull-kubernetes-e2e-gce](https://k8s-testgrid.appspot.com/presubmits-kubernetes-blocking#pull-kubernetes-e2e-gce&include-filter-by-regex=Aggregator%20Should%20be%20able%20to%20support%20th)
* [ci-kubernetes-e2e-gci-gce](https://k8s-testgrid.appspot.com/sig-api-machinery#gce)

## Triage link
More nuanced logs and details can be found at https://storage.googleapis.com/k8s-gubernator/triage/index.html?sig=api-machinery&job=ci-kubernetes-e2e-gci-gce&test=Aggregator%20Should%20be%20able%20to%20support%20the%201.7%20Sample%20AP

# Error
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/apimachinery/aggregator.go:76
gave up waiting for apiservice wardle to come up successfully
Expected error:
    <*errors.errorString | 0xc420131150>: {
        s: "timed out waiting for the condition",
    }
    timed out waiting for the condition
not to have occurred
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/apimachinery/aggregator.go:337

/priority failing-test
/priority important-soon
/kind flake
/sig-api-machinery
@kubernetes/sig-api-machinery-bugs 
/assign @liggitt 

@liggitt can you please help triage this issue?

### PR DESCRIPTION
**Pull request by fedebongio**
**What this PR does / why we need it**:
It upgrades the parameters on the sample-apiserver generation scripts to generate a 1.10 version of the sample api sever

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #
Fixes #63622

**Special notes for your reviewer**:

Hope this time I don't mess it up

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by neolit123** [comment]
better to add a release note here?
/hold
/kind bug
/kind failing-test

**Comment by yliaog** [comment]
the built image seems to have problem, when used in the e2e test, got the following errors:

standard_init_linux.go:178: exec user process caused "exec format error"

and the container cannot be started.

**Comment by ixdy** [comment]
sure enough, `gcr.io/kubernetes-e2e-test-images/sample-apiserver:1.10` is just a single image (not a manifest list), and it only contains arm64 binaries, which seems wrong. @fedebongio did you push this image? if so, how did you build/push it?

**Comment by fedebongio** [comment]
TL;DR: I've uploaded the wrong image (arm) instead of amd by mistake.

They way this images were generated in the past went through some refactors and transitions lately thanks to @dims and we are now in a much better place.

As far as I was able to gather history from the creators of the test and the image, the generation and upload was always manual (and was kind of broken until recently Dims made it work / reorganized again). 

When running make WHAT=sample-apiserver it will generate the images for the different architectures. I've mistakenly uploaded (docker tag ... and then docker push ...) the arm one instead of the expected amd one. Fixed that today. 

If there was a different automated / better way of uploading these automated images I didn't know / couldnt find anybody who knew, and apologize! Please point me to it for future updates.

**Comment by ixdy** [comment]
The way to build/push the e2e test images nowadays is to `cd` into `test/images` and then run `make all-container` followed by `make all-push`.

**Comment by ixdy** [comment]
yeah, I left off the `WHAT=sample-apiserver` instruction, but you figured that out.

it should build and push images for all architectures and then publish a manifest list as well.

**Comment by fedebongio** [comment]
Added the s390 line to BASEIMAGE. In parallel, was able to do make container-all and make push-all succesfully to gcr.

**Comment by spiffxp** [comment]
/lgtm
/hold cancel
/release-note-none
Given that this is a test change I don't see this as deserving of a release note, if you disagree @neolit123 (or someone else) let me know, I can edit the description to include one
